sw/device/lib/base/log_dv.h

```diff
@@ -0,0 +1,34 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef OPENTITAN_SW_DEVICE_LIB_LOG_DV_H_
+#define OPENTITAN_SW_DEVICE_LIB_LOG_DV_H_
+
+/**
+ * Stringify stuff.
+ *
+ * The following set of macros enable a const entity to be stringified.
+ */
+#define LOG_DV_STRINGIFY(a) _LOG_DV_STRINGIFY_INNER_(a)
+#define LOG_DV_STRINGIFY_INNER_(a) #a
+
+/**
+ * The following meta-programming helps arrive at the number of va_args
+ *
+ * Currently, there is no way of knowing the number of var args at compile time.
+ * This information is needed for the 'backdoor' logging to work. The following
+ * meta-programming enables us to get that information as long as the number of
+ * var args passed is <=32.
+ */
+#define ___VA_NARGS(xf, x0, x1, x2, x3, x4, x5, x6, x7, x8, x9, x10, x11, x12, \
+                    x13, x14, x15, x16, x17, x18, x19, x20, x21, x22, x23,     \
+                    x24, x25, x26, x27, x28, x29, x30, x31, n, ...)            \
+  n
+#define __VA_NARGS(...) ___VA_NARGS(__VA_ARGS__)
+#define _VA_NARGS(...)                                                        \
+  __VA_NARGS(__VA_ARGS__, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, \
+             18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0)
```

Comment:
These names are still using leading `_A`. Consider giving them more descriptive names; the varying number of underscores is not readable.